src/comparator/service_comparator.py

```diff
@@ -119,6 +119,41 @@ def _compareRpcMethods(
                 FindingContainer.addFinding(
                     FindingCategory.METHOD_PAGINATED_RESPONSE_CHANGE, "", msg, True
                 )
+            # 6.8 The method_signature annotation is changed.
+            signatures_original = self._get_signatures(method_original)
+            signatures_update = self._get_signatures(method_update)
+            self._compare_method_signatures(signatures_original, signatures_update)
+
+    def _get_signatures(self, method: MethodDescriptorProto):
+        """Return the signature defined for this method."""
+        return method.options.Extensions[client_pb2.method_signature]
+
+    def _compare_method_signatures(self, signatures_original, signatures_update):
+        def _filter_fields(signatures):
+            fields = [
+                field.strip() for sig in signatures for field in sig.split(",") if field
+            ]
+            return fields
+
+        # Flatten the method_signatures fields.
+        # For example: ['content, error'] to ['content', 'error']
+        fields_original = _filter_fields(signatures_original)
+        fields_update = _filter_fields(signatures_update)
+        if len(fields_original) > len(fields_update):
+            FindingContainer.addFinding(
+                FindingCategory.METHOD_SIGNATURE_CHANGE,
+                "",
+                "An existing method_signature is removed.",
+                True,
+            )
+        for old_sig, new_sig in zip(fields_original, fields_update):
```

Comment:
It seems this doesn't cover the case that user switched the signatures order, for instance, from "a,b,c" to "c,b,a".
```
>>> a=["a","b","c"]
>>> b=["c","b","a"]
>>> for x,y in zip(a,b):
...     print(f"{x}, {y}")
... 
a, c
b, b
c, a
```

Reply:
If `field_original` and `fields_update` are sets, then it works.
```
>>> a=set({"a","b","c"})
>>> b=set({"c","b","a"})
>>> for x,y in zip(a,b):
...     print(f"{x}, {y}")
... 
b, b
c, c
a, a
```

Reply:
Unfortunately, changing the order of existing method_signatures is considered breaking change. From [AIP](https://google.aip.dev/client-libraries/4232) It says `Order matters here`. But I agree converting list to set is an efficient way to compare non-ordered elements. Thanks!

Reply:
I see.